plugins/modules/mount.py

```diff
@@ -602,6 +642,17 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
     except IOError:
         module.fail_json(msg="Cannot close file %s" % mntinfo_file)
 
+    return lines
+
+
+def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
+    """Gather mount information"""
+
+    lines = _get_mount_info(module, mntinfo_file)
+    # Keep same behavior than before
+    if lines is None:
+        return
```

Comment:
The function `get_linux_mount` used to implement directly the `_get_mount_info` logic. Since the new function `_is_same_mount_src` also needed the mount info, I centralized this process in `_get_mount_info` to avoid code redundancy.

The behavior of `get_linux_mounts` is unchanged (return nothing if `IOError` is raised when opening the file, fail module if `IOError` is raised when closing the file).

Reply:
@NeodymiumFerBore I have tested this FIX and it worked first run on *BSD, Solars, and Linux.
However, when I tried the second run, it only succeeded on Linux host, and other OS has been failed with the following error:

```shell
fatal: [freebsd]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/local/bin/python3.8"}, "changed": false, "msg": "Unable to retrieve mount info from '/proc/self/mountinfo'"}
fatal: [netbsd]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/pkg/bin/python3.9"}, "changed": false, "msg": "Unable to retrieve mount info from '/proc/self/mountinfo'"}
fatal: [openbsd]: FAILED! => {"ansible_facts": {"discovered_interpreter_python": "/usr/local/bin/python3.8"}, "changed": false, "msg": "Unable to retrieve mount info from '/proc/self/mountinfo'"}
fatal: [solaris]: FAILED! => {"changed": false, "msg": "Unable to retrieve mount info from '/proc/self/mountinfo'"}
```

Even if the `ephemeral` option only works on Linux, I think we need to take care of other OS.

Reply:
## Why integration tests fail on *BSD and Unix as of now

@saito-hideki I don't know what values you used for the *src* parameter, but as of now, it might success if you are using network sources (currently, the tests fail on *bsd and solaris due to the way they handle loop devices, I'm implementing a fix for that).

## Why manual tests might work first run, but not second

If it failed on second run (and I assume you didn't run the integration tests, but did manual tests), I think it is due to the function `_is_same_mount_src`. It uses the function `_get_mount_info` without checking if the system is Linux, and `/proc/self/mountinfo` can only be reliably used on Linux systems. As does `/sys/block/loop${x}/loop/backing_file`.

## Note about checking the backing file of loop devices

This might be useless to try to get the backing file of a loop device on some systems. When we format/mount a file, Linux manipulates this file a block device by **implicitly** creating/using a loop device.
Solaris does not support this, and we need to **explicitly** create a loop file interface for the file (with `lofiadm`). I have to do more tests about this for OpenBSD, FreeBSD and NetBSD.

In this version of my PR, when using `ephemeral` *state*, `mount.py` tries to verify if *path* is a mountpoint, and if it finds a loop device (`/dev/loop[0-9]+`), it tries to retrieve the backing file through `/sys/block/loop${x}/loop/backing_file`. This is required to make the `ephemeral` *state* compatible with mounting files (implicit loop devices).

On systems that do not implicitly creates this loop device (such as Solaris), **it might be useless to try to find the backing file of a loop device**, as if a user tries to mount a file, he will first manually find/create the loop device, as he would on a shell.

I working on this.

Reply:
TLDR: Im implementing a fix so the `ephemeral` *state* works fine on *BSD, Solaris and Linux. It might take some time and a lot of testing.

Reply:
Moved from `/proc/self/mountinfo` to `mount -v`, which has a close behavior on Linux, Solaris and *BSD systems.
Also implemented integration tests for Solaris/*BSD